src/librustc/ty/context.rs

```diff
@@ -387,6 +369,7 @@ impl<'a, 'gcx, 'tcx> Deref for TyCtxt<'a, 'gcx, 'tcx> {
 }
 
 pub struct GlobalCtxt<'tcx> {
+    global_arenas: &'tcx CtxtArenas<'tcx>,
```

Comment:
A rename to `GlobalArenas` might make more sense. In fact, a `GlobalInterners` type would help too, I think `layout` isn't the only one, and `stability` also is global-only.